tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ComputeTimeDerivative.cpp

```diff
@@ -0,0 +1,580 @@
+  using type = Scalar<DataVector>;
+};
+
+struct PackagedVar1 : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct PackagedVar2 : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim, Frame::Inertial>;
+};
+
+template <size_t Dim, SystemType system_type>
+struct TimeDerivative {
+  struct Var3Squared : db::SimpleTag {
+    using type = Scalar<DataVector>;
+  };
+  using temporary_tags = tmpl::list<Var3Squared>;
+  using argument_tags = tmpl::list<Var1, Var2<Dim>, Var3>;
+
+  // Conservative system
+  static void apply(
+      const gsl::not_null<Scalar<DataVector>*> dt_var1,
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> dt_var2,
+
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> flux_var1,
+      const gsl::not_null<tnsr::IJ<DataVector, Dim, Frame::Inertial>*>
+          flux_var2,
+
+      const gsl::not_null<Scalar<DataVector>*> square_var3,
+
+      const Scalar<DataVector>& var1,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& var2,
+      const Scalar<DataVector>& var3) noexcept {
+    get(*square_var3) = square(get(var3));
+
+    // Set source terms
+    get(*dt_var1) = get(*square_var3);
+    for (size_t d = 0; d < Dim; ++d) {
+      dt_var2->get(d) = get(var3) * d;
+    }
+
+    // Set fluxes
+    for (size_t i = 0; i < Dim; ++i) {
+      flux_var1->get(i) = square(get(var1)) * var2.get(i);
+      for (size_t j = 0; j < Dim; ++j) {
+        flux_var2->get(i, j) = var2.get(i) * var2.get(j) * get(var1);
+        if (i == j) {
+          flux_var2->get(i, j) += cube(get(var1));
+        }
+      }
+    }
+  }
+
+  // Nonconservative system
+  static void apply(
+      const gsl::not_null<Scalar<DataVector>*> dt_var1,
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> dt_var2,
+
+      const gsl::not_null<Scalar<DataVector>*> square_var3,
+
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& d_var1,
+      const tnsr::iJ<DataVector, Dim, Frame::Inertial>& d_var2,
+
+      const Scalar<DataVector>& var1,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& var2,
+      const Scalar<DataVector>& var3) noexcept {
+    get(*square_var3) = square(get(var3));
+
+    // Set source terms and nonconservative products
+    get(*dt_var1) = get(*square_var3);
+    for (size_t d = 0; d < Dim; ++d) {
+      get(*dt_var1) -= var2.get(d) * d_var1.get(d);
+      dt_var2->get(d) = get(var3) * d;
+      for (size_t i = 0; i < Dim; ++i) {
+        dt_var2->get(d) -= get(var1) * var2.get(i) * d_var2.get(i, d);
+      }
+    }
+  }
+
+  // Mixed system
+  static void apply(
+      const gsl::not_null<Scalar<DataVector>*> dt_var1,
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> dt_var2,
+
+      const gsl::not_null<tnsr::IJ<DataVector, Dim, Frame::Inertial>*>
+          flux_var2,
+
+      const gsl::not_null<Scalar<DataVector>*> square_var3,
+
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& d_var1,
+
```

Comment:
I personally like having these in order to differentiate visually what the different quantities are for. Eg time derivative/sources, fluxes, temporaries, partial derivatives, extra arguments. What do you think?

Reply:
I dislike the blank lines because to me they look like a formatting error, not like actual information.

To make it more like information, on option would be to intersperse comments like "// the following arguments are ___", before each group of arguments. I would be happy with that approach if you'd like to keep the groups separated 🙂